sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SymbolicExecution/BaseExplodedGraph.cs

```diff
@@ -1,442 +1,442 @@
+        #endregion
+
+        #region Visit*
+
+        protected abstract void VisitBinaryBranch(BinaryBranchBlock binaryBranchBlock, ExplodedGraphNode node);
+
+        protected abstract void VisitInstruction(ExplodedGraphNode node);
+
+        protected virtual void VisitBranchBlock(ExplodedGraphNode node, ProgramPoint programPoint)
+        {
+            var newProgramState = node.ProgramState.PopValue();
+            newProgramState = CleanStateAfterBlock(newProgramState, node.ProgramPoint.Block);
+            EnqueueAllSuccessors(programPoint.Block, newProgramState);
+        }
+
+        protected virtual void VisitBranchBlock(BranchBlock branchBlock, ExplodedGraphNode node)
+        {
+            var newProgramState = node.ProgramState;
+            if (IsValueConsumingStatement(branchBlock.BranchingNode))
+            {
+                newProgramState = newProgramState.PopValue();
+            }
+            newProgramState = CleanStateAfterBlock(newProgramState, branchBlock);
+            EnqueueAllSuccessors(branchBlock, newProgramState);
+        }
+
+        protected virtual void VisitSingleSuccessorBinaryBranch(BinaryBranchingSimpleBlock block, ExplodedGraphNode node)
+        {
+            SymbolicValue sv;
+            var programState = node.ProgramState.PopValue(out sv);
+
+            foreach (var newProgramState in sv.TrySetConstraint(BoolConstraint.True, programState))
+            {
+                OnConditionEvaluated(block.BranchingInstruction, evaluationValue: true);
+                var nps = newProgramState.PushValue(SymbolicValue.True);
+                EnqueueNewNode(new ProgramPoint(block.SuccessorBlock), nps);
+            }
+
+            foreach (var newProgramState in sv.TrySetConstraint(BoolConstraint.False, programState))
+            {
+                OnConditionEvaluated(block.BranchingInstruction, evaluationValue: false);
+                var nps = newProgramState.PushValue(SymbolicValue.False);
+                EnqueueNewNode(new ProgramPoint(block.SuccessorBlock), nps);
+            }
+        }
+
+        protected virtual void VisitSimpleBlock(SimpleBlock block, ExplodedGraphNode node)
+        {
+            var newProgramState = CleanStateAfterBlock(node.ProgramState, block);
+
+            var jumpBlock = block as JumpBlock;
+            if (jumpBlock != null &&
+                IsValueConsumingStatement(jumpBlock.JumpNode))
+            {
+                newProgramState = newProgramState.PopValue();
+            }
+
+            EnqueueAllSuccessors(block, newProgramState);
+        }
+
+        protected abstract bool IsValueConsumingStatement(SyntaxNode jumpNode);
+
+        protected ProgramState CleanStateAfterBlock(ProgramState programState, Block block)
+        {
+            var liveVariables = lva.GetLiveOut(block)
+                .Union(nonInDeclarationParameters); // LVA excludes out and ref parameters
+
+            // TODO: Remove the IFieldSymbol check when SLVS-1136 is fixed
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)